### DISCUSSION TITLE
No "exports" main defined in E:\control-bench\node_modules\jose\package.json

### DISCUSSION THREAD

**Discussion by zhushenhang** [panva/jose#137] [Q&A] [answered]
hi,dear friends,sorry for my English.
i got an error when I upgraded from version 1.10.1 to 3.5.0.
my code :
```
const { JWK } = require('jose');
const key = JWK.generateSync('oct');
const util = require('./util');

const common = {
  key,
  util
};

module.exports = common;

```
has the key generation been modified?

**Comment by panva** [comment, marked as answer]
> i got an error when I upgraded from version 1.10.1 to 3.5.0.

You're jumping two semver major versions.

> has the key generation been modified?

yes, that's why major version has been bumped.

**Reply by zhushenhang** [reply]
thanks，can I still generate keys in version 3.5.0? 
what should I do to generate the key?

**Reply by panva** [reply]
Are you expecting an answer other than "please refer to the documentation"?

**Reply by zhushenhang** [reply]
yes.
i read the JWK documentation, and i found that there is no API for generating keys. 
so, I want to know how to randomly generate a "kid" as before.
like this code.
```
OctKey {
  kid: 'CdB99ks7oPjzFs6dQmcp7OTu2Y3ynwHP8aXh4bcQ544',
  kty: 'oct'
}
```

**Reply by panva** [reply]
It's all in the docs.

**Reply by panva** [reply]
https://github.com/panva/jose/blob/master/docs/functions/_jwk_thumbprint_.calculatethumbprint.md#readme

**Reply by panva** [reply]
https://github.com/panva/jose/blob/master/docs/functions/_util_generate_key_pair_.generatekeypair.md#readme

**Reply by panva** [reply]
https://github.com/panva/jose/blob/master/docs/functions/_util_generate_secret_.generatesecret.md#readme

**Reply by zhushenhang** [reply]
oh，I thought there were only a few: Parsing (JWK to KeyLike)，Conversion (KeyLike to JWK)，Thumbprints，EmbeddedJWK, 
thank you so much.

**Comment by kdhttps** [comment]
Adding example. It may save someones' time :)
```js
const { generateKeyPair, exportJWK, calculateJwkThumbprint } = require('jose') // jose@4.1.4

async function generateJWKS () {
  const { publicKey, privateKey } = await generateKeyPair('RS256')

  const privateJwk = await exportJWK(privateKey)
  console.log('JWKS json with public and private key : ', privateJwk)

  const publicJwk = await exportJWK(publicKey)
  console.log('JWKS json with only public key : ', publicJwk)

  const kid = await calculateJwkThumbprint(publicJwk)
  console.log('kid : ', kid)
}

generateJWKS()
```
